### PR TITLE
ARROW-2066: [Python] Document using pyarrow with Azure Blob Store

### PR DESCRIPTION
**Pull request by rjrussell77**
Original question:

https://github.com/apache/arrow/issues/1510

Improvement story:

https://issues.apache.org/jira/browse/ARROW-2066

### CODE REVIEW

**Review by rjrussell77 — COMMENTED** [review]
@wesm Can you review this?  (I had difficulty with getting the formatting down. Adding a sub-bullet list item caused the parent bullet to become italicized.  Eventually omitted the sub-bullet.)

**Comment by rrussell39** [comment]
@xhochy Uwe - can you review?

File: python/doc/source/parquet.rst [hunk 1]
```diff
@@ -237,3 +237,42 @@ throughput:
...
+   import tempfile
+
+   account_name = '...'
+   account_key = '...'
+   container_name = '...'
+   parquet_file = 'mysample.parquet'
+
+   block_blob_service = BlockBlobService(account_name=account_name, account_key=account_key)
+   with tempfile.TemporaryFile() as fp:
+      block_blob_service.get_blob_to_stream(container_name=container_name, blob_name=parquet_file, stream=fp)
```
Comment (xhochy) [inline python/doc/source/parquet.rst]: This should actually work without a temporary file, see the implementation in `simplekv`: https://github.com/mbr/simplekv/blob/master/simplekv/net/azurestore.py#L74

**Review by xhochy — CHANGES_REQUESTED** [review]

File: python/doc/source/parquet.rst (hunk 1 above)
Comment (rrussell39) [reply python/doc/source/parquet.rst]: Ok - I'll take a look.  Thanks.

File: python/doc/source/parquet.rst [hunk 2]
```diff
@@ -237,3 +237,40 @@ throughput:
...
+   account_name = '...'
+   account_key = '...'
+   container_name = '...'
+   parquet_file = 'mysample.parquet'
+
+   block_blob_service = BlockBlobService(account_name=account_name, account_key=account_key)
+   byte_stream = io.BytesIO()
+   block_blob_service.get_blob_to_stream(container_name=container_name, blob_name=parquet_file, stream=byte_stream)
+   pd = pq.read_table(source=byte_stream).to_pandas()
+   pd.head(10)
```
Comment (rjrussell77) [inline python/doc/source/parquet.rst]: @xhochy Good feedback - I replaced the temp file buffer with BytesIO stream instead.

File: python/doc/source/parquet.rst [hunk 3]
```diff
@@ -237,3 +237,44 @@ throughput:
...
+   block_blob_service = BlockBlobService(account_name=account_name, account_key=account_key)
+   try:
+      block_blob_service.get_blob_to_stream(container_name=container_name, blob_name=parquet_file, stream=byte_stream)
+      pd = pq.read_table(source=byte_stream).to_pandas()
+      pd.head(10)
+   except Exception as err:
+      print("Error: {0}".format(err))
+   finally:
+      byte_stream.close()
+
```
Comment (rjrussell77) [inline python/doc/source/parquet.rst]: Added try/except/finally block to ensure closure of the stream

File: python/doc/source/parquet.rst [hunk 4]
```diff
@@ -237,3 +237,44 @@ throughput:
...
+   account_name = '...'
+   account_key = '...'
+   container_name = '...'
+   parquet_file = 'mysample.parquet'
+
+   block_blob_service = BlockBlobService(account_name=account_name, account_key=account_key)
+   try:
+      block_blob_service.get_blob_to_stream(container_name=container_name, blob_name=parquet_file, stream=byte_stream)
+      pd = pq.read_table(source=byte_stream).to_pandas()
```
Comment (xhochy) [inline python/doc/source/parquet.rst]: The result is typically written into a variable called `df` whereas `pd` is the abbreviation you use when you import pandas (`import pandas as pd`)

File: python/doc/source/parquet.rst [hunk 5]
```diff
@@ -237,3 +237,44 @@ throughput:
...
+      pd.head(10)
```
Comment (xhochy) [inline python/doc/source/parquet.rst]: Better replace this with `# Do work on DF …`

File: python/doc/source/parquet.rst [hunk 6]
```diff
@@ -237,3 +237,44 @@ throughput:
...
+   except Exception as err:
```
Comment (xhochy) [inline python/doc/source/parquet.rst]: Please don't catch exceptions like this, just let it throw.

File: python/doc/source/parquet.rst (hunk 3 above)
Comment (xhochy) [reply python/doc/source/parquet.rst]: Can you add this comment to the code? That will also be helpful for the reader later.

File: python/doc/source/parquet.rst (hunk 5 above)
Comment (rjrussell77) [reply python/doc/source/parquet.rst]: How about, `# Do work on df` (lower case)?

File: python/doc/source/parquet.rst [hunk 7]
```diff
@@ -237,3 +237,43 @@ throughput:
...
+
+   block_blob_service = BlockBlobService(account_name=account_name, account_key=account_key)
+   try:
+      block_blob_service.get_blob_to_stream(container_name=container_name, blob_name=parquet_file, stream=byte_stream)
+      df = pq.read_table(source=byte_stream).to_pandas()
+      # Do work on df ...
+   finally:
+      # Add finally block to ensure closure of the stream
+      byte_stream.close()
+
```
Comment (rjrussell77) [inline python/doc/source/parquet.rst]: @xhochy Ok, I've responded to your last set of feedback.  How are we looking now?

**Review by xhochy — APPROVED** [review]
+1, thank you for writing this up. This will be very helpful for new users.

**Comment by xhochy** [comment]
@rjrussell77 do you have an Apache JIRA id so I could assign https://issues.apache.org/jira/browse/ARROW-2066 to you?

**Comment by rjrussell77** [comment]
@xhochy Regarding your question about my Apache JIRA id - try this: rob_PTL

**Comment by wesm** [comment]
Thanks @rjrussell77 -- I added you as a contributor and assigned the issue to you

**Pull request closed without merging**